test/integration/test_raw_api.py

```diff
@@ -39,595 +24,577 @@
+def authorize_raw_api(raw_api):
+    if (application_key_id := os.environ.get('B2_TEST_APPLICATION_KEY_ID')) is None:
         pytest.fail('B2_TEST_APPLICATION_KEY_ID is not set.')
 
-    application_key = os.environ.get('B2_TEST_APPLICATION_KEY')
-    if application_key is None:
+    if (application_key := os.environ.get('B2_TEST_APPLICATION_KEY')) is None:
         pytest.fail('B2_TEST_APPLICATION_KEY is not set.')
 
-    print()
+    realm = os.environ.get('B2_TEST_ENVIRONMENT', 'production')
+    realm_url = REALM_URLS.get(realm, realm)
+    auth_dict = raw_api.authorize_account(realm_url, application_key_id, application_key)
+    return auth_dict
 
-    try:
-        raw_api = B2RawHTTPApi(B2Http())
-        raw_api_test_helper(raw_api, not dont_cleanup_old_buckets)
-    except Exception:
-        traceback.print_exc(file=sys.stdout)
-        pytest.fail('test_raw_api failed')
 
+class RawApiIntegrationTestBase:
+    raw_api: B2RawHTTPApi
+    auth_dict: dict
+    api_url: str
+    account_auth_token: str
+    account_id: str
+    sse_b2_aes: EncryptionSetting
+    
+    @classmethod
+    def setup_class(cls):
+        # very verbose http debug:
+        # import http.client; http.client.HTTPConnection.debuglevel = 1
+        cls.raw_api = B2RawHTTPApi(B2Http())
+        cls.sse_b2_aes = EncryptionSetting(
+            mode=EncryptionMode.SSE_B2,
+            algorithm=EncryptionAlgorithm.AES256,
+        )
 
-def authorize_raw_api(raw_api):
-    application_key_id = os.environ.get('B2_TEST_APPLICATION_KEY_ID')
-    if application_key_id is None:
-        print('B2_TEST_APPLICATION_KEY_ID is not set.', file=sys.stderr)
-        sys.exit(1)
+        cls.authorize_account()
 
-    application_key = os.environ.get('B2_TEST_APPLICATION_KEY')
-    if application_key is None:
-        print('B2_TEST_APPLICATION_KEY is not set.', file=sys.stderr)
-        sys.exit(1)
+    @classmethod
+    def authorize_account(cls):
+        print("b2_authorize_account")
+        cls.auth_dict = authorize_raw_api(cls.raw_api)
 
-    realm = os.environ.get('B2_TEST_ENVIRONMENT', 'production')
-    realm_url = REALM_URLS.get(realm, realm)
-    auth_dict = raw_api.authorize_account(realm_url, application_key_id, application_key)
-    return auth_dict
+        missing_capabilities = set(ALL_CAPABILITIES) - {'readBuckets', 'listAllBucketNames'
+                                                        } - set(cls.auth_dict['allowed']['capabilities'])
+        assert not missing_capabilities, 'it appears that the raw_api integration test is being run with a non-full key. Missing capabilities: {}'.format(
+            missing_capabilities,
+        )
 
+        cls.account_id = cls.auth_dict['accountId']
+        cls.account_auth_token = cls.auth_dict['authorizationToken']
+        cls.api_url = cls.auth_dict['apiUrl']
+        cls.download_url = cls.auth_dict['downloadUrl']
 
-def raw_api_test_helper(raw_api, should_cleanup_old_buckets):
```

Comment:
You're right that this refactor could be a different PR. I decided I wanted to have a clear split between bucket and non-bucket tests, and since this way of splitting makes more sense in the context of what is done in this PR (this  refactor prob wouldn't have been done otherwise), I decided to do it here. At the end of this PR I'll rebase my commits and have a separate commit for this refactor.